docs/config-file.md

```diff
@@ -12,30 +12,84 @@ export default async function () {
 }
 ```
 
-This function must return an object with the following shape:
+This function must return an object with protocol specific configuration which would look like this: 
 
+
+```js
+export default async function () {
+  return {
+    websocket: {},
+    mqtt: {},
+    cluster: {}
+  }
+}
+
+```
+
+Here is an example of the `glee.config.js` file for refence: 
 ```js
 export default async function () {
   return {
     websocket: {
-      httpServer: customServer, // A custom HTTP server of your own.
-      adapter: 'native', // Default. Can also be 'socket.io' or a reference to a custom adapter.
-      port: process.env.PORT,
+      server: {
+        httpServer: customServer, // A custom HTTP server of your own.
+        adapter: "native", // Default. Can also be 'socket.io' or a reference to a custom adapter.
+        port: process.env.PORT,
+      },
+      client: {
+        authentication: {
+          http: () => {
+            const token = process.env.TOKEN
+            return token
+          }
+        }
+      }
     },
     cluster: {
-      adapter: 'redis',
-      name: 'cluster', // Default. Name of your cluster.
-      url: 'redis://localhost:6379', // Server URL used by adapter for clustering
+      adapter: "redis",
+      name: "cluster", // Default. Name of your cluster.
+      url: "redis://localhost:6379", // Server URL used by adapter for clustering
+    },
+    mqtt: {
+      authentication: {
+        cert: () => fs.readFileSync('./cert')
+        clientId: '123',
+        userPassword: {
+          username: 'user1',
+          password: 'pass12'
+        }
+      }
     }
-  }
+  };
 }
 ```
+Every protocol has different configuration needs so each protocol has unique configurations:
+
+### Websockets
+|Field|Description|
+|--|--|
+|websocket.server|Websocket server specific configuration|
+|websocet.client| Websocket client specific configuration|
```

Comment:
```suggestion
|websocket.client| Websocket client-specific configuration|
```

Can you also please move it below to group it with the other `websocket.client` configs? It's hard to read as it is.